engine/hid/src/hid_private.h

```diff
@@ -54,16 +58,17 @@ namespace dmHID
     {
         Context();
 
-        Gamepad             m_Gamepads[MAX_GAMEPAD_COUNT];
-        Keyboard            m_Keyboards[MAX_KEYBOARD_COUNT];
-        Mouse               m_Mice[MAX_MOUSE_COUNT];
-        TouchDevice         m_TouchDevices[MAX_TOUCH_DEVICE_COUNT];
-        TextPacket          m_TextPacket;
-        MarkedTextPacket    m_MarkedTextPacket;
-        AccelerationPacket  m_AccelerationPacket;
-        DMHIDGamepadFunc    m_GamepadConnectivityCallback;
-        void*               m_GamepadConnectivityUserdata;
-        void*               m_NativeContext;
+        Gamepad            m_Gamepads[MAX_GAMEPAD_COUNT];
+        Keyboard           m_Keyboards[MAX_KEYBOARD_COUNT];
+        Mouse              m_Mice[MAX_MOUSE_COUNT];
+        TouchDevice        m_TouchDevices[MAX_TOUCH_DEVICE_COUNT];
+        TextPacket         m_TextPacket;
+        MarkedTextPacket   m_MarkedTextPacket;
+        AccelerationPacket m_AccelerationPacket;
+        DMHIDGamepadFunc   m_GamepadConnectivityCallback;
+        void*              m_GamepadConnectivityUserdata;
+        void*              m_NativeContext;
+        void*              m_NativeContextUserData;
```

Comment:
I added the context user data here so that I don't have to store more static pointers in the native hid impl, but perhaps we can use the "nativecontext" instead? Not sure what it's used for?

Reply:
I think the idea behind it was to support any platform specific data, e.g. a HHandle or similar.